src/Action_GIST.h

```diff
@@ -6,6 +6,17 @@
 #ifdef CUDA
 #include "cuda_kernels/GistCudaSetup.cuh"
 #endif
+
+#include "DataSet_3D.h"
```

Comment:
We may want to see if some of these includes can be replaced with forward declares.